src/rl/proof_tree.py

```diff
@@ -81,9 +81,20 @@ def __str__(self) -> str:
 NumberOfBacktracks: {self.num_of_backtracks}
 PossibleFailedPaths: {self.possible_failed_paths}
 AdditionalInfo: {self.additional_info}
+"""         
+
+            if self.language == ProofAction.Language.ISABELLE:
+                all_proof_steps = "\n".join(lines) if len(lines) > 0 else ""
+                return f"""{self.lemma_name}
+{all_proof_steps}
+{proof_metadata}
```

Comment:
output Isabelle proofs differently, since they may not have a beginning and ending token